tests/test_processing_stack.py

```diff
@@ -154,7 +168,56 @@ def should_successfully_run_dataset_version_creation_process_with_multiple_asset
     metadata_copy_job_result = None
     asset_copy_job_result = None
 
-    with S3Object(
+    with Dataset() as dataset, S3Object(
+        file_object=json_dict_to_file_object(
+            {
+                **deepcopy(MINIMAL_VALID_STAC_CATALOG_OBJECT),
+                STAC_ID_KEY: dataset.dataset_prefix,
+                STAC_TITLE_KEY: dataset.title,
+                STAC_LINKS_KEY: [
+                    {
+                        STAC_REL_KEY: STAC_REL_ROOT,
+                        STAC_HREF_KEY: f"../{CATALOG_FILENAME}",
+                        STAC_TITLE_KEY: ROOT_CATALOG_TITLE,
+                        STAC_TYPE_KEY: STAC_MEDIA_TYPE_JSON,
+                    },
+                    {
+                        STAC_REL_KEY: STAC_REL_PARENT,
+                        STAC_HREF_KEY: f"../{CATALOG_FILENAME}",
+                        STAC_TITLE_KEY: ROOT_CATALOG_TITLE,
+                        STAC_TYPE_KEY: STAC_MEDIA_TYPE_JSON,
+                    },
+                ],
+            }
+        ),
+        bucket_name=Resource.STORAGE_BUCKET_NAME.resource_name,
+        key=f"{dataset.dataset_prefix}/{CATALOG_FILENAME}",
+    ), S3Object(
+        file_object=json_dict_to_file_object(
+            {
+                **deepcopy(MINIMAL_VALID_STAC_CATALOG_OBJECT),
+                STAC_ID_KEY: ROOT_CATALOG_ID,
+                STAC_DESCRIPTION_KEY: ROOT_CATALOG_DESCRIPTION,
+                STAC_TITLE_KEY: ROOT_CATALOG_TITLE,
+                STAC_LINKS_KEY: [
+                    {
+                        STAC_REL_KEY: STAC_REL_ROOT,
+                        STAC_HREF_KEY: f"./{CATALOG_FILENAME}",
+                        STAC_TITLE_KEY: ROOT_CATALOG_TITLE,
+                        STAC_TYPE_KEY: STAC_MEDIA_TYPE_JSON,
+                    },
+                    {
+                        STAC_REL_KEY: STAC_REL_CHILD,
+                        STAC_HREF_KEY: f"./{dataset.dataset_prefix}/{CATALOG_FILENAME}",
+                        STAC_TITLE_KEY: dataset.title,
+                        STAC_TYPE_KEY: STAC_MEDIA_TYPE_JSON,
+                    },
+                ],
+            }
+        ),
+        bucket_name=Resource.STORAGE_BUCKET_NAME.resource_name,
+        key=CATALOG_FILENAME,
```

Comment:
Will modify these to call Datasets Endpoint instead of creating the catalogs directly.